lib/src/local_storage.dart

```diff
@@ -113,3 +116,37 @@ class HiveLocalStorage extends LocalStorage {
         .put(supabasePersistSessionKey, persistSessionString);
   }
 }
+
+/// local storage to store pkce flow code verifier.
+class HiveGotrueAsyncStorage extends GotrueAsyncStorage {
```

Comment:
You have stated [here](https://github.com/supabase/supabase-flutter/issues/287#issuecomment-1336246821) that you are open for using shared preferences as well. Is there a reason to pick hive again?

Reply:
Thanks for asking! No particular reason, but mostly only because we are currently using Hive, and using SharedPreferences will add dependencies, but would you prefer to just start using SharedPreferences right now?

We could embed a script similar to [yours for migrating hive path](https://github.com/supabase/supabase-flutter/issues/294#issuecomment-1334944076) to migrate the auth persistence string stored in hive to sharedPreferences and just start using shared preferences right now. 

I don't know what I was thinking during [this](https://github.com/supabase/supabase-flutter/issues/294#issuecomment-1334689312) discussion, but no need to wait until v2 to do this for sure. In fact, if we would to make the migration, we should do it early, and keep the script over multiple versions, right?